Add SBOM provenance verification to SLSA guide (#1561)
fluxcd/website: content/en/flux/security/slsa-assessment.md

GitHub Workflow Repository: fluxcd/source-controller
GitHub Workflow Ref: refs/tags/v1.0.0
```

### Flux artifacts

The provenance of the Flux release artifacts (binaries, container images, SBOMs, deploy manifests) published on
GitHub can be verified using the official [SLSA verifier tool](https://github.com/slsa-framework/slsa-verifier).

### Example

> We will be using the Flux SBOM file for this example, but these instructions can be used for all Flux release artifacts.
> 
> First, we download the release artifacts from GitHub:
> 
> ```shell
> FLUX_VER=2.0.1 && \
> gh release download v${FLUX_VER} -R=fluxcd/flux2 -p="*"
> ```
> 
> Using the `provenance.intoto.jsonl` file, we can verify the provenance attestation of the Flux SBOM:
> 
> ```console
> $ slsa-verifier verify-artifact --provenance-path provenance.intoto.jsonl --source-uri github.com/fluxcd/flux2 --source-tag v${FLUX_VER} flux_${FLUX_VER}_sbom.spdx.json
> Verified signature against tlog entry index 27066821 at URL: https://rekor.sigstore.dev/api/v1/log/entries/24296fb24b8ad77ac2d2dc6381ec7f1f04d991344771214c5fb5861621dbd9da6f0551f806cbf609
> Verified build using builder https://github.comslsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.7.0 at commit 9b3162495ce1b99b1fcdf137c553f543eafe3ec7
> Verifying artifact flux_2.0.1_sbom.spdx.json: PASSED
> PASSED: Verified SLSA provenance
> ```

Reviewer: ```suggestion
In this example we use the Flux SBOM file, but the instructions can be used for all artifacts included with the Flux release.

First, download the release artifacts from GitHub:

```shell
FLUX_VER=2.0.1 && \
gh release download v${FLUX_VER} -R=fluxcd/flux2 -p="*"
```

Using the `provenance.intoto.jsonl` file, verify the provenance attestation of the Flux SBOM (`flux_<version>_sbom.spdx.json`):

```console
$ slsa-verifier verify-artifact --provenance-path provenance.intoto.jsonl --source-uri github.com/fluxcd/flux2 --source-tag v${FLUX_VER} flux_${FLUX_VER}_sbom.spdx.json
Verified signature against tlog entry index 27066821 at URL: https://rekor.sigstore.dev/api/v1/log/entries/24296fb24b8ad77ac2d2dc6381ec7f1f04d991344771214c5fb5861621dbd9da6f0551f806cbf609
Verified build using builder https://github.comslsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.7.0 at commit 9b3162495ce1b99b1fcdf137c553f543eafe3ec7
Verifying artifact flux_2.0.1_sbom.spdx.json: PASSED
PASSED: Verified SLSA provenance
```
```

Reviewer: Argh, GitHub still does not like nested ``` blocks. But you should get the idea when you try to edit my comment.
